### PR TITLE
Release Notes for December 2023 Prerelease SDK

### PR DESCRIPTION
**Pull request by mikehoffms**
Rendered article section for review:
https://review.learn.microsoft.com/microsoft-edge/webview2/release-notes?branch=pr-en-us-2968&tabs=dotnetcsharp#12102261-prerelease

AB#48097065

### CODE REVIEW

File: microsoft-edge/webview2/release-notes.md [hunk 1]
```diff
@@ -138,6 +138,311 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
...
+
+* [ICoreWebView2GetProcessExtendedInfosCompletedHandler](/microsoft-edge/webview2/reference/win32/icorewebview2getprocessextendedinfoscompletedhandler?view=webview2-1.0.2261-prerelease&preserve-view=true)
+
+* [ICoreWebView2ProcessExtendedInfo](/microsoft-edge/webview2/reference/win32/icorewebview2processextendedinfo?view=webview2-1.0.2261-prerelease&preserve-view=true)
+    * [ICoreWebView2ProcessExtendedInfo::get_ProcessInfo](/microsoft-edge/webview2/reference/win32/icorewebview2processextendedinfo?view=webview2-1.0.2261-prerelease&preserve-view=true#get_processinfo)
+    * [ICoreWebView2ProcessExtendedInfo::get_AssociatedFrameInfos](/microsoft-edge/webview2/reference/win32/icorewebview2processextendedinfo?view=webview2-1.0.2261-prerelease&preserve-view=true#get_associatedframeinfos)
+
+* [ICoreWebView2ProcessExtendedInfoCollection](/microsoft-edge/webview2/reference/win32/icorewebview2processextendedinfocollection?view=webview2-1.0.2261-prerelease&preserve-view=true)
+
+* [ICoreWebView2_20](/microsoft-edge/webview2/reference/win32/icorewebview2_20?view=webview2-1.0.2210.57&preserve-view=true)
```
Comment (mikehoffms) [inline microsoft-edge/webview2/release-notes.md:345]: error of some sort

Comment (mikehoffms) [inline microsoft-edge/webview2/release-notes.md:345]:
```suggestion
* [ICoreWebView2_20](https://learn.microsoft.com/microsoft-edge/webview2/reference/win32/icorewebview2_20?view=webview2-1.0.2261-prerelease&preserve-view=true)
    * [ICoreWebView2_20::get_FrameId](https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2_20?view=webview2-win32-preview&branch=pr-en-us-76#get_frameid)<!--no put-->
```

File: microsoft-edge/webview2/release-notes.md [hunk 2]
```diff
@@ -138,6 +138,311 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
...
+
+* [ICoreWebView2Frame5](/microsoft-edge/webview2/reference/win32/icorewebview2frame5?view=webview2-1.0.2261-prerelease&preserve-view=true)
+    * [ICoreWebView2Frame5::get_FrameId](/microsoft-edge/webview2/reference/win32/icorewebview2frame5?view=webview2-1.0.2261-prerelease&preserve-view=true#get_frameid)
+
+* [ICoreWebView2FrameInfo2](/microsoft-edge/webview2/reference/win32/icorewebview2frameinfo2?view=webview2-1.0.2261-prerelease&preserve-view=true)
+    * [ICoreWebView2FrameInfo2::get_FrameId](/microsoft-edge/webview2/reference/win32/icorewebview2frameinfo2?view=webview2-1.0.2261-prerelease&preserve-view=true#get_frameid)
+    * [ICoreWebView2FrameInfo2::get_FrameKind](/microsoft-edge/webview2/reference/win32/icorewebview2frameinfo2?view=webview2-1.0.2261-prerelease&preserve-view=true#get_framekind)
+    * [ICoreWebView2FrameInfo2::get_ParentFrameInfo](/microsoft-edge/webview2/reference/win32/icorewebview2frameinfo2?view=webview2-1.0.2261-prerelease&preserve-view=true#get_parentframeinfo)
+    * [ICoreWebView2FrameInfo2::Invoke](/microsoft-edge/webview2/reference/win32/icorewebview2frameinfo2?view=webview2-1.0.2261-prerelease&preserve-view=true#invoke)
```
Comment (mikehoffms) [inline microsoft-edge/webview2/release-notes.md:354]: @peiche-jessica @victorhuangwq - no "Invoke" member near 
https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2frameinfo2?view=webview2-win32-preview&branch=pr-en-us-76

File: microsoft-edge/webview2/release-notes.md [hunk 3]
```diff
@@ -138,6 +138,311 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
...
+<!-- ---------- -->
+* Added the `ExecuteScriptWithResult` method, which will provide exception information if the executed script failed, and additionally adds a new method `TryGetResultAsString` that tries to get the script execution result as a string rather than as JSON in order to make it more convenient to interact with string results.
+
+<!-- todo: make sure lists are sync'd across tabs -->
+
+##### [.NET/C#](#tab/dotnetcsharp)
+
+<!-- todo: test links -->
+
+* [CoreWebView2.CreateFromComICoreWebView2 Method](/dotnet/api/microsoft.web.webview2.core.corewebview2.createfromcomicorewebview2?view=webview2-dotnet-1.0.2261-prerelease&preserve-view=true)
```
Comment (mikehoffms) [inline microsoft-edge/webview2/release-notes.md:370]: check whether this CreateFromComICoreWebView2 member actually exists for .NET

it's not in WinRT, near 
https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2?branch=pr-en-us-46&view=webview2-winrt-preview

File: microsoft-edge/webview2/release-notes.md [hunk 4]
```diff
@@ -138,6 +138,311 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
...
+
+* [CoreWebView2ScriptException Class](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2scriptexception?view=webview2-winrt-1.0.2261-prerelease&preserve-view=true)
+    * [CoreWebView2ScriptException.ColumnNumber Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2scriptexception?view=webview2-winrt-1.0.2261-prerelease&preserve-view=true#columnnumber)
+    * [CoreWebView2ScriptException.Message Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2scriptexception?view=webview2-winrt-1.0.2261-prerelease&preserve-view=true#message)
+    * [CoreWebView2ScriptException.ToJson Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2scriptexception?view=webview2-winrt-1.0.2261-prerelease&preserve-view=true#tojson)
+    * [LineNumber Enum](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/linenumber?view=webview2-winrt-1.0.2261-prerelease&preserve-view=true)
```
Comment (mikehoffms) [inline microsoft-edge/webview2/release-notes.md:404]: not enum but property on fooException?

File: microsoft-edge/webview2/release-notes.md [hunk 5]
```diff
@@ -138,6 +138,311 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
...
+##### [WinRT/C#](#tab/winrtcsharp)
+
+<!-- todo: test links -->
+
+* [CoreWebView2 Class](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2?view=webview2-winrt-1.0.2261-prerelease&preserve-view=true)
+    * [CoreWebView2.ExecuteScriptWithResultAsync Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2?view=webview2-winrt-1.0.2261-prerelease&preserve-view=true#executescriptwithresultasync)
+
+* [CoreWebView2ExecuteScriptResult Class](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2executescriptresult?view=webview2-winrt-1.0.2261-prerelease&preserve-view=true)
+    * [CoreWebView2ExecuteScriptResult.Exception Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2executescriptresult?view=webview2-winrt-1.0.2261-prerelease&preserve-view=true#exception)
+    * [CoreWebView2ExecuteScriptResult.FrameId Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2executescriptresult?view=webview2-winrt-1.0.2261-prerelease&preserve-view=true#frameid)
```
Comment (mikehoffms) [inline microsoft-edge/webview2/release-notes.md:467]: Remove CoreWebView2ExecuteScriptResult.**FrameId**?  
It's not listed in
https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2executescriptresult?view=webview2-winrt-preview&branch=pr-en-us-46

@peiche-jessica @victorhuangwq

File: microsoft-edge/webview2/release-notes.md [hunk 6]
```diff
@@ -138,6 +138,387 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
...
+Win32 rendered PR:
+https://review.learn.microsoft.com/microsoft-edge/webview2/reference/win32/icorewebview2frameinfo2?view=webview2-win32-preview&branch=pr-en-us-76
+-->
+
+##### [.NET/C#](#tab/dotnetcsharp)
+
+<!-- todo: test links -->
+
+* `CoreWebView2` Class:
+    * [CoreWebView2.CreateFromComICoreWebView2 Method](/dotnet/api/microsoft.web.webview2.core.corewebview2.createfromcomicorewebview2?view=webview2-dotnet-1.0.2261-prerelease&preserve-view=true)
```
Comment (mikehoffms) [inline microsoft-edge/webview2/release-notes.md:433]: Delete CreateFromComICoreWebView2?  It's not listed in WinRT or Win32 tab, not found in WinRT near 
https://review.learn.microsoft.com/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2?view=webview2-winrt-preview&branch=pr-en-us-46
@peiche-jessica @victorhuangwq

File: microsoft-edge/webview2/release-notes.md (hunk 4 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md]: redid as a property on fooException

File: microsoft-edge/webview2/release-notes.md (hunk 1 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md]: latest Files Changed tab Markdown has 2261:
* [ICoreWebView2_20](/microsoft-edge/webview2/reference/win32/icorewebview2_20?view=webview2-1.0.2261-prerelease&preserve-view=true)
    * [ICoreWebView2_20::get_FrameId](/microsoft-edge/webview2/reference/win32/icorewebview2_20?view=view=webview2-1.0.2261-prerelease&preserve-view=true#get_frameid)

File: microsoft-edge/webview2/release-notes.md (hunk 6 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md]: Comment is Outdated; surrounding Markdown not the latest.
Latest commits/ fresh Files Changed tab: https://github.com/MicrosoftDocs/edge-developer/pull/2968/files

File: microsoft-edge/webview2/release-notes.md (hunk 5 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md]: Comment is Outdated, which means the surrounding Markdown is not the latest.
Latest commits/ fresh Files Changed tab: https://github.com/MicrosoftDocs/edge-developer/pull/2968/files

File: microsoft-edge/webview2/release-notes.md (hunk 3 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md]: Comment is Outdated, which means the surrounding Markdown is not the latest.
Latest commits/ fresh Files Changed tab: https://github.com/MicrosoftDocs/edge-developer/pull/2968/files

File: microsoft-edge/webview2/release-notes.md (hunk 2 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md]: Comment is Outdated, which means the surrounding Markdown is not the latest.
Latest commits/ fresh Files Changed tab: https://github.com/MicrosoftDocs/edge-developer/pull/2968/files

File: microsoft-edge/webview2/release-notes.md [hunk 7]
```diff
@@ -138,6 +138,307 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
...
+    * [ICoreWebView2FrameInfo2::get_FrameId](/microsoft-edge/webview2/reference/win32/icorewebview2frameinfo2?view=webview2-1.0.2261-prerelease&preserve-view=true#get_frameid)<!--no put-->
+    * [ICoreWebView2FrameInfo2::get_FrameKind](/microsoft-edge/webview2/reference/win32/icorewebview2frameinfo2?view=webview2-1.0.2261-prerelease&preserve-view=true#get_framekind)<!--no put-->
+    * [ICoreWebView2FrameInfo2::get_ParentFrameInfo](/microsoft-edge/webview2/reference/win32/icorewebview2frameinfo2?view=webview2-1.0.2261-prerelease&preserve-view=true#get_parentframeinfo)<!--no put-->
+    * [ICoreWebView2FrameInfo2::Invoke](/microsoft-edge/webview2/reference/win32/icorewebview2frameinfo2?view=webview2-1.0.2261-prerelease&preserve-view=true#invoke)<!-- todo: no "Invoke" member near https://review.learn.microsoft.com/microsoft-edge/webview2/reference/win32/icorewebview2frameinfo2?view=webview2-win32-preview&branch=pr-en-us-76 - for WinRT, no Invoke member, either: https://review.learn.microsoft.com/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2frameinfo?view=webview2-winrt-preview&branch=pr-en-us-46 -->
```
[the remark this answers is not in the document]
Comment (victorhuangwq) [reply microsoft-edge/webview2/release-notes.md:353]: @mikehoffms please remove the invoke function. It was a mistake in the diffs

File: microsoft-edge/webview2/release-notes.md [hunk 8]
```diff
@@ -138,6 +138,307 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
...
+
+* `CoreWebView2` Class:
+    * [CoreWebView2.ExecuteScriptWithResultAsync Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2?view=webview2-winrt-1.0.2261-prerelease&preserve-view=true#executescriptwithresultasync)
+
+* [CoreWebView2ExecuteScriptResult Class](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2executescriptresult?view=webview2-winrt-1.0.2261-prerelease&preserve-view=true)
+    * [CoreWebView2ExecuteScriptResult.Exception Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2executescriptresult?view=webview2-winrt-1.0.2261-prerelease&preserve-view=true#exception)
+    * [CoreWebView2ExecuteScriptResult.FrameId Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2executescriptresult?view=webview2-winrt-1.0.2261-prerelease&preserve-view=true#frameid)<!--todo: FrameId is not listed in Ref docs, delete it? https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2executescriptresult?view=webview2-winrt-preview&branch=pr-en-us-46 -->
```
Comment (victorhuangwq) [inline microsoft-edge/webview2/release-notes.md:391]: remove this as well, delete this

File: microsoft-edge/webview2/release-notes.md (hunk 5 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md]: from chat from VH: 
the invoke function shouldn't be there
I added the comments on where to delete
similarly for [CoreWebView2ExecuteScriptResult.FrameId Property]

File: microsoft-edge/webview2/release-notes.md (hunk 2 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md]: from chat from VH: 
the invoke function shouldn't be there

File: microsoft-edge/webview2/release-notes.md (hunk 8 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md:391]: deleted FrameId

File: microsoft-edge/webview2/release-notes.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md:353]: removed invoke.  older example for context: invoke is on a handler:
```
* [ICoreWebView2ExperimentalLaunchingExternalUriSchemeEventHandler::Invoke](/microsoft-edge/webview2/reference/win32/icorewebview2experimentallaunchingexternalurischemeeventhandler?view=webview2-1.0.1724-prerelease&preserve-view=true#invoke)
```

File: microsoft-edge/webview2/release-notes.md (hunk 6 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md]: latest has a hidden todo comment

File: microsoft-edge/webview2/release-notes.md (hunk 5 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md]: deleted CoreWebView2ExecuteScriptResult.FrameId

File: microsoft-edge/webview2/release-notes.md (hunk 3 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md]: this PR comment is probably dup.
latest .md has at this method: ""

File: microsoft-edge/webview2/release-notes.md (hunk 2 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md]: deleted ICoreWebView2FrameInfo2::Invoke.   invoke method is on win32 handler types

File: microsoft-edge/webview2/release-notes.md (hunk 3 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md]: this PR comment is a dup of https://github.com/MicrosoftDocs/edge-developer/pull/2968/files/ab68de770929dc8ae3e6a0f812d6271725795e83#r1424431294
closing present comment

File: microsoft-edge/webview2/release-notes.md (hunk 6 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md]: closed the dup comment https://github.com/MicrosoftDocs/edge-developer/pull/2968/files/36830a1396281068c49d41752f487c094658d651#r1423280779

Comment (victorhuangwq) [reply microsoft-edge/webview2/release-notes.md]: I agree that we should delete it. It's not related to execute script, and is not found in winrt either

Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md]: deleted

File: microsoft-edge/webview2/release-notes.md [hunk 9]
```diff
@@ -370,6 +370,322 @@ The following APIs have been promoted to Stable and are now included in this Rel
 <!-- end of Dec 2023 Release SDK -->
 
 
+<!-- ====================================================================== -->
+## 121.0.2261-prerelease
```
Comment (mikehoffms) [inline microsoft-edge/webview2/release-notes.md:374]: todo: Update version number throughout section
@victorhuangwq @peiche-jessica

File: microsoft-edge/webview2/release-notes.md [hunk 10]
```diff
@@ -370,6 +370,322 @@ The following APIs have been promoted to Stable and are now included in this Rel
 <!-- end of Dec 2023 Release SDK -->
 
 
+<!-- ====================================================================== -->
+## 121.0.2261-prerelease
+
+<!-- todo: Version # TBD; update throughout section -->
+
+Release Date: December 13, 2023<!-- todo: finalize date -->
```
Comment (mikehoffms) [inline microsoft-edge/webview2/release-notes.md:378]: todo: Finalize date
@victorhuangwq @peiche-jessica

File: microsoft-edge/webview2/release-notes.md [hunk 11]
```diff
@@ -370,6 +370,322 @@ The following APIs have been promoted to Stable and are now included in this Rel
...
+<!-- ------------------------------ -->
+#### Experimental APIs
+
+
+<!-- ---------- -->
+* Added the Runtime selection feature to support more prerelease testing and flighting scenarios. Developers can specify `ReleaseChannels` to choose which channels are searched for during environment creation, and `ChannelSearchKind` to select a search order.
+
+##### [.NET/C#](#tab/dotnetcsharp)
+
+<!-- todo: test links -->
```
Comment (mikehoffms) [inline microsoft-edge/webview2/release-notes.md:394]: todo: test .NET links in the six .NET tabs in this Prerelease h2 section

**Review by victorhuangwq — APPROVED** [review]
Apart from this link issue:

image


I think it should work. Approve tentatively after change of version number

File: microsoft-edge/webview2/release-notes.md [hunk 12]
```diff
@@ -370,6 +370,323 @@ The following APIs have been promoted to Stable and are now included in this Rel
...
+    * [ICoreWebView2ScriptException::get_LineNumber](/microsoft-edge/webview2/reference/win32/icorewebview2scriptexception?view=webview2-1.0.2261-prerelease&preserve-view=true#get_linenumber)<!--no put-->
+    * [ICoreWebView2ScriptException::get_Message](/microsoft-edge/webview2/reference/win32/icorewebview2scriptexception?view=webview2-1.0.2261-prerelease&preserve-view=true#get_message)<!--no put-->
+    * [ICoreWebView2ScriptException::get_Name](/microsoft-edge/webview2/reference/win32/icorewebview2scriptexception?view=webview2-1.0.2261-prerelease&preserve-view=true#get_name)<!--no put-->
+    * [ICoreWebView2ScriptException::get_ToJson](/microsoft-edge/webview2/reference/win32/icorewebview2scriptexception?view=webview2-1.0.2261-prerelease&preserve-view=true#get_tojson)<!--no put-->
+
+---
+
+
+<!-- ---------- -->
+* The `CoreWebView2.CreateFromComICoreWebView2` method wraps an existing `ICoreWebView2` object in a `CoreWebView2` instance, to enable .NET developers to interact with an control that was created in C++.
```
[the remark this answers is not in the document]
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md:654]: changed an to a

**Pull request closed without merging**